File: SOLID/Liskov.java

class Animal {
    public void hacerSonido() {
        System.out.println("Haciendo sonido.");
    }
}

class Perro extends Animal {
    public void hacerSonido() {
        System.out.println("Guau!");
    }
}

public class Liskov {
    public static void main(String[] args) {


        Animal animal = new Perro();
        animal.hacerSonido();


    }
}

File: SOLID/Liskov.cpp
#include <iostream>
using namespace std;

class Animal {
public:
    virtual void hacerSonido() {
        cout << "Haciendo sonido." << endl;
    }
};

class Perro : public Animal {
public:
    void hacerSonido() override {
        cout << "Guau!" << endl;
    }
};

int main() {
    Animal* animal = new Perro();
    animal->hacerSonido();

    delete animal;
    return 0;
}
